api/spec/src/types.tsp

```diff
@@ -78,6 +78,22 @@ model UniqueResource {
   key: Key;
 }
 
+/**
+ * IDResource is a resouce with an ID.
+ */
+// NOTE: this can be used to have a type, that we can later replace with the expanded type if needed without
+// breaking api compatibility
+@friendlyName("IDResource")
+model IDResource {
```

Comment:
What's the value of doing IDResource? Is the resource type changing for the fields where it's used?

Reply:
This btw reminds me of GraphQL's global IDs where every resource's ID is globally unique. Usually it's a concatenation of the type and ID .

Reply:
This is just an API compatibility issue. This allows the response to be like this:

```
{
   "subscription": {
      "subscription": {
         "id": ...
      },
      "phase": {
          "id": ...
      }
...
}
```

This way if we want to extend the API with extending the objects later, we can do that without needing to break the API.